lib/fog/vsphere/requests/compute/get_folder.rb

```diff
@@ -23,7 +23,7 @@ def get_raw_folder(path, datacenter_name_or_obj, type)
                end
 
           valid_types = %w[vm network datastore host]
-          raise ArgumentError, "#{type} is unknown" if type.blank?
+          raise ArgumentError, "#{type} is unknown" if type.empty?
```

Comment:
`x.blank?` is a rails alias for `x.nil? || x.empty?` so you should also check for that here